docs/DeveloperGuide.md

```diff
@@ -312,9 +312,17 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should be able to hold up to 1000 students without a noticeable sluggishness in performance for typical usage.
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4. The system should respond within two seconds
+5. The system should be usable by someone who's tech illiterate
+6. The system should work on both 32-bit and 54-bit environments
+7. The system should be backward compatible with data produced by earlier versions of the systems
```

Comment:
Idt we should enforce this, especially if we want to change stuff in the future, thoughts @tensaida @yeojunjie @LuYiting0913?